File: programmers/lv2/math/당구 연습.py
def solution(m, n, startX, startY, balls):
    answer = []
    
    for ball in balls:
        x, y = ball
        candidates = [[x, (-1) * y], [(-1) * x, y], [x, y + 2 * (n - y)], [x + 2 * (m - x), y]]
        a, b, c, d = startX == x and startY > y, startY == y and startX > x, startX == x and startY < y, startY == y and startX < x
        check = [a, b, c, d]
        min_num = float('inf')
        for i, coordinate in enumerate(candidates):
            if check[i]:
                continue
            min_num = min(min_num, pow(startX - coordinate[0], 2) + pow(startY - coordinate[1], 2))
        answer.append(min_num)
        
    return answer

File: programmers/lv2/연습문제/당구 연습.java
class Solution {
    public int[] solution(int m, int n, int startX, int startY, int[][] balls) {
        int[] answer = new int[balls.length];
        
        for(int i = 0; i < balls.length; i++) {
            int min = Integer.MAX_VALUE;
            
            int tx = balls[i][0], ty = balls[i][1];
            if(startY != ty || startX < tx) {
                min = Math.min(min, (int) Math.pow(tx + startX, 2) + (int) Math.pow(Math.abs(ty- startY), 2));
            }
            if(startY != ty || startX > tx) {
                min = Math.min(min, (int) Math.pow((m - tx) + (m - startX), 2) + (int) Math.pow(Math.abs(ty - startY), 2));
            }
            if(startX != tx || startY < ty) {
                min = Math.min(min, (int) Math.pow(ty + startY, 2) + (int) Math.pow(Math.abs(tx - startX), 2));
            }
            if(startX != tx || startY > ty) {
                min = Math.min(min, (int) Math.pow((n - ty) + (n - startY), 2) + (int) Math.pow(Math.abs(tx - startX), 2));
            }
            
            answer[i] = min;
        }
        
        return answer;
    }
}
